lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2251,7 +2255,7 @@ class VarDeclUsageChecker : public ASTWalker {
     
   // After we have scanned the entire region, diagnose variables that could be
   // declared with a narrower usage kind.
-  ~VarDeclUsageChecker() override;
+  ~VarDeclUsageChecker() override __attribute__((optnone));
```

Comment:
Is there a clang bug here?

Reply:
No this was an oversight. I will remove it.